auto/generate_test_runner.rb

```diff
@@ -143,25 +143,38 @@ def find_tests(source)
 
       if @options[:use_param_tests] && !arguments.empty?
         args = []
-        type_and_args = arguments.split(/TEST_(CASE|RANGE)/)
+        type_and_args = arguments.split(/TEST_(CASE|RANGE|MATRIX)/)
         for i in (1...type_and_args.length).step(2)
-          if type_and_args[i] == "CASE"
+          case type_and_args[i]
+          when "CASE"
             args << type_and_args[i + 1].sub(/^\s*\(\s*(.*?)\s*\)\s*$/m, '\1')
-            next
-          end
 
-          # RANGE
-          args += type_and_args[i + 1].scan(/(\[|<)\s*(-?\d+.?\d*)\s*,\s*(-?\d+.?\d*)\s*,\s*(-?\d+.?\d*)\s*(\]|>)/m).map do |arg_values_str|
-            exclude_end = arg_values_str[0] == '<' && arg_values_str[-1] == '>'
-            arg_values_str[1...-1].map do |arg_value_str|
-              arg_value_str.include?('.') ? arg_value_str.to_f : arg_value_str.to_i
-            end.push(exclude_end)
-          end.map do |arg_values|
-            Range.new(arg_values[0], arg_values[1], arg_values[3]).step(arg_values[2]).to_a
-          end.reduce(nil) do |result, arg_range_expanded|
-            result.nil? ? arg_range_expanded.map { |a| [a] } : result.product(arg_range_expanded)
-          end.map do |arg_combinations|
-            arg_combinations.flatten.join(', ')
+          when "RANGE"
+            args += type_and_args[i + 1].scan(/(\[|<)\s*(-?\d+.?\d*)\s*,\s*(-?\d+.?\d*)\s*,\s*(-?\d+.?\d*)\s*(\]|>)/m).map do |arg_values_str|
+              exclude_end = arg_values_str[0] == '<' && arg_values_str[-1] == '>'
+              arg_values_str[1...-1].map do |arg_value_str|
+                arg_value_str.include?('.') ? arg_value_str.to_f : arg_value_str.to_i
+              end.push(exclude_end)
+            end.map do |arg_values|
+              Range.new(arg_values[0], arg_values[1], arg_values[3]).step(arg_values[2]).to_a
+            end.reduce(nil) do |result, arg_range_expanded|
+              result.nil? ? arg_range_expanded.map { |a| [a] } : result.product(arg_range_expanded)
+            end.map do |arg_combinations|
+              arg_combinations.flatten.join(', ')
+            end
+
+          when "MATRIX"
+            single_arg_regex_string = /(?:(?:"(?:\\"|[^\\])*?")+|(?:'\\?.')+|(?:[^\s\]\["'\,]|\[[\d\S_-]+\])+)/.source
```

Comment:
IMO, I split parts of single argument regex for readability & easily improving every regex part independently.

Reply:
Yeah, I noticed that in your PR, for some reason separating the elements makes it visualize how the regex will process the command.
I normally only do it if I need to reuse parts of the query, so I only have one place to maintain